fvm/errors/errors.go

```diff
@@ -114,6 +114,39 @@ func HandleRuntimeError(err error) error {
 		return NewUnknownFailure(err)
 	}
 
+	errs := []error{runErr}
+	var parentErr errors.ParentError
+	if As(err, &parentErr) {
+		errs = parentErr.ChildErrors()
+	}
+
+	codedErrs := make([]CodedError, 0)
+	for _, err := range errs {
+		coded, isUnknown := findImportantCodedError(err)
+		if isUnknown {
+			// ignore uncoded errors
+			continue
+		}
+
+		if coded.Code().IsFailure() {
+			return WrapCodedError(
+				coded.Code(),
+				err,
```

Comment:
Maybe it would be better to wrap the whole err here (not the err form the for loop, but the err from the outer scope ), so we don't loose information.